src/user_agent.rs

```diff
@@ -1,226 +1,339 @@
+                    Some(loc) => Some(&a[loc.0..loc.1]),
+                    _ => None,
+                },
+                gemstash: match bl.get(11) {
+                    Some(loc) => Some(&a[loc.0..loc.1]),
+                    _ => None,
+                },
+            })
+        } else if self.ruby_pattern.captures_read(rl, a).is_some() {
+            return Some(UserAgent {
+                agent_name: Some("rubygems"),
+                agent_version: match rl.get(1) {
+                    Some(loc) => Some(&a[loc.0..loc.1]),
+                    _ => panic!("parse failed on {:?}", a),
+                },
+                bundler: None,
+                rubygems: match rl.get(1) {
+                    Some(loc) => Some(&a[loc.0..loc.1]),
+                    _ => panic!("parse failed on {:?}", a),
+                },
+                ruby: match rl.get(3) {
+                    Some(loc) => Some(&a[loc.0..loc.1]),
+                    _ => None,
+                },
+                platform: match rl.get(2) {
+                    Some(loc) => Some(&a[loc.0..loc.1]),
+                    _ => None,
+                },
+                command: None,
+                jruby: None,
+                truffleruby: None,
+                options: None,
+                ci: None,
+                gemstash: match rl.get(4) {
+                    Some(loc) => Some(&a[loc.0..loc.1]),
+                    _ => None,
+                },
+            });
+        } else if self.gem_pattern.captures_read(gl, a).is_some() {
+            return Some(UserAgent {
+                agent_name: Some("gems"),
+                agent_version: match gl.get(1) {
+                    Some(loc) => Some(&a[loc.0..loc.1]),
+                    _ => panic!("parse failed on {:?}", a),
+                },
+                bundler: None,
+                rubygems: None,
+                ruby: None,
+                platform: None,
+                command: None,
+                jruby: None,
+                truffleruby: None,
+                options: None,
+                ci: None,
+                gemstash: None,
+            });
+        } else if self
+            .generic_pattern
+            .captures_read(&mut capture_locations.generic_captures, a)
+            .is_some()
+        {
+            return Some(UserAgent {
+                agent_name: capture_locations
+                    .generic_captures
+                    .get(1)
+                    .map(|m| &a[m.0..m.1]),
+                agent_version: capture_locations
+                    .generic_captures
+                    .get(2)
+                    .map(|m| &a[m.0..m.1]),
+                ..Default::default()
+            });
+        } else {
+            return None;
+        }
+    }
+}
+
+pub fn serialize_user_agent<S>(ua: &str, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    lazy_static! {
+        static ref USER_AGENT_PARSER: ParseCtx = ParseCtx::new();
```

Comment:
@iximeow any recommendations for avoiding these two `lazy_static!`s inside serde serialize methods?